src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -84,7 +85,8 @@ object WdlExpression {
   }
 }
 
-case class WdlExpression(ast: AstNode) {
+case class WdlExpression(ast: AstNode) extends WdlValue {
+  val wdlType = WdlExpressionType
```

Comment:
If you look at how `wdlType` is used I don't think it can be a `type`, but I could be wrong.